compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3003,76 +3011,315 @@ private BLangExpression createConditionFromMatchPatterns(List<BLangMatchPattern>
+        matchStmtsForPattern.clear();
+        return ASTBuilderUtil.createIfElseStmt(pos, condition, successBody, null);
+    }
+
+    private BLangExpression createConditionForWildCardMatchPattern(BLangWildCardMatchPattern wildCardMatchPattern,
+                                                                   BLangSimpleVarRef matchExprVarRef) {
+
+        boolean condition;
+        if (wildCardMatchPattern.matchesAll) {
+            condition = true;
+        } else {
+            condition = types.isAssignable(matchExprVarRef.type, symTable.anyType);
+        }
+        return ASTBuilderUtil.createLiteral(wildCardMatchPattern.pos, symTable.booleanType, condition);
+    }
+
+    private BLangExpression createConditionForConstMatchPattern(BLangConstPattern constPattern,
+                                                                BLangSimpleVarRef matchExprVarRef) {
+
+        return createBinaryExpression(constPattern.pos, matchExprVarRef, constPattern.expr);
+    }
+
+    private BLangExpression createConditionForVarBindingPatternMatchPattern(BLangVarBindingPatternMatchPattern
+                                                                                    varBindingPatternMatchPattern,
+                                                                            BLangSimpleVarRef matchExprVarRef) {
+
+        BLangBindingPattern bindingPattern = varBindingPatternMatchPattern.getBindingPattern();
+
+        switch (bindingPattern.getKind()) {
+            case CAPTURE_BINDING_PATTERN:
+                BLangCaptureBindingPattern captureBindingPattern = (BLangCaptureBindingPattern) bindingPattern;
+                DiagnosticPos captureBindingPatternPos = captureBindingPattern.pos;
+                BLangSimpleVariable captureBindingPatternVar =
+                        ASTBuilderUtil.createVariable(captureBindingPatternPos,
+                                captureBindingPattern.getIdentifier().getValue(), matchExprVarRef.type,
+                                matchExprVarRef, captureBindingPattern.symbol);
+                captureBindingPattern.symbol.type = matchExprVarRef.type;
+                BLangSimpleVariableDef captureBindingPatternVarDef =
+                        ASTBuilderUtil.createVariableDef(captureBindingPatternPos, captureBindingPatternVar);
+                matchStmtsForPattern.add(captureBindingPatternVarDef);
+
+                return ASTBuilderUtil.createLiteral(captureBindingPattern.pos, symTable.booleanType, true);
+        }
+        // If some patterns are not implemented, those should be detected before this phase
+        // TODO : Remove this after all patterns are implemented
+        return null;
+    }
+
+    private BLangExpression createConditionForErrorMatchPattern(BLangErrorMatchPattern errorMatchPattern,
+                                                                BLangSimpleVarRef matchExprVarRef) {
+
+        BType matchPatternType = errorMatchPattern.type;
```

Comment:
Method seems to be too long, it might be better if we can try to refactor

Reply:
Changed